ibm_mq/datadog_checks/ibm_mq/collectors/queue_metric_collector.py

```diff
@@ -107,9 +107,14 @@ def _discover_queues(self, queue_manager, mq_pattern_filter):
                     self.warning("Error discovering queue: %s", e)
             else:
                 for queue_info in response:
-                    queue = to_string(queue_info[pymqi.CMQC.MQCA_Q_NAME]).strip()
-                    self.log.debug("Discovered queue: %s", queue)
-                    queues.append(queue)
+                    queue = queue_info.get(pymqi.CMQC.MQCA_Q_NAME, None)
+                    if queue is not None:
+                        queue_name = to_string(queue).strip()
+                        self.log.debug("Discovered queue: %s", queue_name)
+                        queues.append(queue_name)
+                    elif queue is None:
```

Comment:
Can this happen, and if so can a queue be literaly an empty string? in which case it won't enter through the if because '' is not None.
Maybe the check should be:
if not queue:
    ....
 else:
    ...
    
 `not queue` is True both if its None or if it's empty

Reply:
That works, I'll update it so that it catches both cases.